torch/_dynamo/utils.py

```diff
@@ -1067,6 +1093,13 @@ def get_stack(self):
             self.tls.stack = []
             return self.tls.stack
 
+    def get_top(self) -> str:
```

Comment:
@jamesjwu This and the contextmanager below is probably the part where I need your input the most.